middleware/src/middleware.go

```diff
@@ -1171,3 +1172,32 @@ func (middleware *Middleware) VerifyAppMiddlewarePairing(channelHash []byte) (bo
 	}
 	return true, nil
 }
+
+// HSMUpdateAvailable checks the AvailableHSMVersion Redis and compares it to the running FW version
+func (middleware *Middleware) HSMUpdateAvailable(hsm *hsm.HSM) (bool, error) {
```

Comment:
hsm param unused? maybe better to pass `hsmFirmware` here instead

Reply:
Good catch, I was previously calling hsm.UpdateFirmware() from here, but then moved it out and forgot to remove this.